src/apps/loan-list/list/loan-list.tsx

```diff
@@ -0,0 +1,159 @@
+    const regex = /^\d{4}-\d{2}-\d{2}$/;
+    // modal query param
+    const modalString = searchParams.get("modal");
+
+    if (modalString && loans) {
+      const found = modalString.toString().match(regex);
+      // If there is a date string in the modal query param
+      if (found) {
+        setDueDateModal(found[0]);
+        // THe loans are filtered with said date string
+        const loansForModal = loans.filter(
+          ({ loanDetails }) => loanDetails.dueDate === dueDateModal
+        );
+        // Amount of renewable loans are determined, used in the ui
+        const amountOfRenewableLoans = loansForModal.filter(
+          ({ isRenewable }) => isRenewable
+        ).length;
+
+        // Loans for modal (the modal shows loans stacked by due date)
+        setLoansModal(loansForModal);
+        setRenewable(amountOfRenewableLoans);
+      }
+    }
+  }, [loans, dueDateModal]);
+
+  useEffect(() => {
+    getLoansV2().then((value) => {
+      if (value) {
+        const listOfDueDates = value.map((a) => a.loanDetails.dueDate);
+        const uniqeListOfDueDates = Array.from(new Set(listOfDueDates));
+
+        // The due dates are used for the stacked materials
+        // The stacked materials view shows materials stacked by
+        // due date, and for this we need a uniqe list of due dates
+        setDueDates(uniqeListOfDueDates);
+
+        // Loans are sorted by due date
+        const sortedByDueDate = value.sort(
+          (objA, objB) =>
+            new Date(objA.loanDetails.dueDate).getTime() -
+            new Date(objB.loanDetails.dueDate).getTime()
+        );
+        setLoans(sortedByDueDate);
+        setAmountOfLoans(sortedByDueDate.length);
+      }
+    });
+  }, []);
+
+  function openModalDueDate(dueDateModalInput: string) {
+    if (loans) {
+      setDueDateModal(dueDateModalInput);
+      setLoansModal(
+        loans.filter(
+          ({ loanDetails }) => loanDetails.dueDate === dueDateModalInput
+        )
+      );
+    }
+  }
+
+  return (
+    <div aria-hidden={loansModal !== null ?? true}>
+      <h1 className="text-header-h1 m-32">{t("loanListTitleText")}</h1>
+      {/* Todo style the below */}
+      <h2 className="text-header-h4 m-32">
+        {t("loanListPhysicalLoansTitleText")} ({amountOfLoans})
+      </h2>
+      <button onClick={() => setView("list")} type="button">
+        <img src={MenuIcon} alt={t("loanListListText")} />
+      </button>
+      <button onClick={() => setView("stacked")} type="button">
+        <img src={VariousIcon} alt={t("loanListStackText")} />
+      </button>
+      <button
+        type="button"
+        aria-describedby={t("loanListRenewMultipleButtonExplanationText")}
+        className="btn-primary btn-medium  btn-filled arrow__hover--right-small"
+      >
+        {t("loanListRenewMultipleButtonText")}
+      </button>
+      {loans && (
+        <div className="list-reservation-container">
+          {view === "stacked" &&
```

Comment:
You could consider creating two sub components called something like: 
LoanListItems and LoanListItemsStacked that is shown depending on the `view`.
That would reduce the size of this component/file.